# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1288778523 -28800
#      Wed Nov 03 18:02:03 2010 +0800
# Node ID 0a8e554494c9d30b5ac1ddae4265403f72679d00
# Parent  f72e17e70378c907c078d60133598fd5f84c6875
__patch__.m: Don't index an empty array; bug introduced by changeset 11175.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -10,16 +10,21 @@ 2010-11-03  Konstantinos Poulios  <logar
 	Now this case is handled in src/graphics.cc .
 
 2010-11-03  Konstantinos Poulios  <logari81@googlemail.com>
 
 	* plot/axis.m: Leave zlim unchanged during "axis tight" on 2D plots
 
 2010-11-03  Ben Abbott <bpabbott@mac.com>
 
+	* plot/private/__patch__.m: Don't index an empty array; bug introduced
+	by changeset 11175.
+
+2010-11-03  Ben Abbott <bpabbott@mac.com>
+
 	* miscellaneous/getappdata.m, miscellaneous/isappdata.m,
 	miscellaneous/rmappdata.m, miscellaneous/setappdata.m:
 	Add new appdata function.
 
 2010-11-01  David Bateman  <dbateman@free.fr>
 
 	* plot/__private__/__contour__.m: Use __go_patch__ rather than patch
 	to avoid the cost of the patch callback functions. Ensure that the
diff --git a/scripts/plot/private/__patch__.m b/scripts/plot/private/__patch__.m
--- a/scripts/plot/private/__patch__.m
+++ b/scripts/plot/private/__patch__.m
@@ -178,23 +178,23 @@ function args = setdata (args)
     faces = args {idx};
   endif
   idx = find (cellfun (@(x) strcmpi (x, "vertices"), args))(end) + 1;
   if (idx > nargs)
     vert = [];
   else
     vert = args {idx};
   endif
-  idx = find (cellfun (@(x) strcmpi (x, "facevertexcdata"), args))(end) + 1;
+  idx = find (cellfun (@(x) strcmpi (x, "facevertexcdata"), args), 1, "last") + 1;
   if (isempty(idx) || idx > nargs)
     fvc = [];
   else
     fvc = args {idx};
   endif
-  idx = find (cellfun (@(x) strcmpi (x, "facecolor"), args))(end) + 1;
+  idx = find (cellfun (@(x) strcmpi (x, "facecolor"), args), 1, "last") + 1;
   if (isempty(idx) || idx > nargs)
     if (!isempty (fvc))
       fc = "flat";
     else
       fc = [0, 1, 0];
     endif
     args = {"facecolor", fc, args{:}};
   else
@@ -222,17 +222,17 @@ function args = setdata (args)
   if (size(fvc, 1) == nc || size (fvc, 1) == 1)
     c = reshape (fvc, [1, size(fvc)]);
   else
     if (size(fvc, 2) == 3)
       c = cat(3, reshape (fvc(idx, 1), size(idx)),
               reshape (fvc(idx, 2), size(idx)),
               reshape (fvc(idx, 3), size(idx)));
     else
-      c = reshape (fvc(idx), size(idx));
+      c = [];
     endif
   endif
   args = {"xdata", x, "ydata", y, "zdata", z, "cdata", c, args{:}};
 endfunction
 
 function args = setvertexdata (args)
   args = delfields (args, {"vertices", "faces", "facevertexcdata"});
   ## Remove the readonly fields as well
